app/models/user.rb

```diff
@@ -5,8 +5,47 @@ class User < ApplicationRecord
          :recoverable, :rememberable, :validatable
 
   validates :name, presence: true, length: { maximum: 20 }
-
   has_many :posts
   has_many :comments, dependent: :destroy
   has_many :likes, dependent: :destroy
+  has_many :friendships, dependent: :destroy
+  has_many :friends, through: :friendships, dependent: :destroy
+  has_many :inverse_friendships, class_name: 'Friendship', foreign_key: 'friend_id', dependent: :destroy
+  scope :all_except, ->(user) { where.not(id: user) }
+
+  def friends
+    friendships.map { |friendship| friendship.friend if friendship.status }.compact
+  end
+
+  def pending_friends
+    friendships.map { |friendship| friendship.friend unless friendship.status }.compact
+  end
+
+  def friend_requests
+    inverse_friendships.map { |friendship| friendship.user unless friendship.status }.compact
+  end
```

Comment:
- [ ] `friend_requests` can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby)